### DISCUSSION TITLE
Support for d2 diagrams (and other elements?)

### DISCUSSION THREAD

**Discussion by nfelber** [PLAYG0N/mkdocs-panzoom#12] [Ideas]
Hi, thank you for this really nice plugin.

It would prove very useful in a project where we have to display large [D2 diagrams](https://d2lang.com) in our documentation rendered using the [MkDocs D2 plugin](https://github.com/landmaj/mkdocs-d2-plugin). I was able to get it to work by simply adding elements with class `"d2"` to the output of `HTMLPage._find_element` and querying the `".d2"` selector in `activate_zoom_pan`.

Since Panzoom is itself very versatile and works with a variety of DOM objects, and since adding a new class selector seems to take very little effort, I was thinking that it might be worth adding support for arbitrary selectors to the plugin (in addition to images and Mermaid charts). If this is not something you are interested in, I would still be happy to make a proper PR for D2 support.

**Comment by PLAYG0N** [comment]
That is indeed a good idea. It kinda never occurred to me that anything other than images and mermaid diagrams would use this :) To avoid boilerplate code I would add the tag to the default ones. I might have to refactor the config to make this easily extensible in order to disable the default ones.

**Reply by nfelber** [reply]
Sounds good :)

Let me know if I can help.

**Reply by PLAYG0N** [reply]
It should work now. Let me know if there are any issues.

**Reply by nfelber** [reply]
I upgraded to the latest release and it seems to be working like a charm. This is fantastic, thank you so much!
